modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -370,13 +372,21 @@ private void onReconnectionSuccessful() {
   }
 
   private void onReconnectionFailed(Throwable exception) {
+    dispatchExceptionNotification(exception);
     LOGGER.error(format("Message source '%s' on flow '%s' could not be reconnected. Will be shutdown. %s",
                         sourceModel.getName(), getLocation().getRootContainerName(), exception.getMessage()),
                  exception);
     shutdown();
     reconnecting.set(false);
   }
 
+  private void dispatchExceptionNotification(Throwable exception) {
```

Comment:
firing notifications is something that the exception handler might do. Rename this method to describe what it's actually doing which is invoking the system handler